modules/file-input/test/spec.coffee

```diff
@@ -16,116 +18,265 @@ describe "File Input", ->
-    should.not.exist( selection.select('input').attr('disabled') )
-    should.not.exist( selection.select('button').attr('disabled') )
-    selection.select('.hx-file-input-selected').classed('hx-text-disabled hx-background-disabled').should.equal(false)
-    fileInput.disabled().should.equal(false)
-
-    fileInput.disabled(true).should.equal(fileInput)
-    selection.select('input').attr('disabled').should.equal('disabled')
-    selection.select('button').attr('disabled').should.equal('disabled')
-    selection.select('.hx-file-input-selected').classed('hx-text-disabled hx-background-disabled').should.equal(true)
-    fileInput.disabled().should.equal(true)
-
-    fileInput.disabled(false).should.equal(fileInput)
-    should.not.exist( selection.select('input').attr('disabled') )
-    should.not.exist( selection.select('button').attr('disabled') )
-    selection.select('.hx-file-input-selected').classed('hx-text-disabled hx-background-disabled').should.equal(false)
-    fileInput.disabled().should.equal(false)
-
-  it 'the fluid api version should return a selection', ->
+    testFileInput undefined, (sel, fileInput) ->
+      should.not.exist( sel.select('input').attr('disabled') )
+      should.not.exist( sel.select('button').attr('disabled') )
+      sel.select('.hx-file-input-selected').classed('hx-text-disabled hx-background-disabled').should.equal(false)
+      fileInput.disabled().should.equal(false)
+
+      fileInput.disabled(true).should.equal(fileInput)
+      sel.select('input').attr('disabled').should.equal('disabled')
+      sel.select('button').attr('disabled').should.equal('disabled')
+      sel.select('.hx-file-input-selected').classed('hx-text-disabled hx-background-disabled').should.equal(true)
+      fileInput.disabled().should.equal(true)
+
+      fileInput.disabled(false).should.equal(fileInput)
+      should.not.exist( sel.select('input').attr('disabled') )
+      should.not.exist( sel.select('button').attr('disabled') )
+      sel.select('.hx-file-input-selected').classed('hx-text-disabled hx-background-disabled').should.equal(false)
+      fileInput.disabled().should.equal(false)
+
+
+  it 'should allow drag and drop by default', ->
+    testFileInput undefined, (sel, fileInput) ->
+      fakeDragEvent = {
+        preventDefault: chai.spy()
+        stopPropagation: chai.spy()
+        dataTransfer: {
+          files: [ fakeFile1 ]
+        }
+      }
+      fakeNodeEvent(sel.node(), 'dragenter')(fakeDragEvent)
+      fakeDragEvent.preventDefault.should.have.been.called.once()
+      fakeDragEvent.stopPropagation.should.have.been.called.once()
+
+      fakeNodeEvent(sel.node(), 'dragover')(fakeDragEvent)
+      fakeDragEvent.preventDefault.should.have.been.called.twice()
+      fakeDragEvent.stopPropagation.should.have.been.called.twice()
+
+      fakeNodeEvent(sel.node(), 'drop')(fakeDragEvent)
+      fakeDragEvent.preventDefault.should.have.been.called.exactly(3)
+      fileInput.value().should.eql([ fakeFile1 ])
```

Comment:
These tests are assuming it's reasonable to mimic the drag events using the 'fake node event' function